src/cart/crt_init.c

```diff
@@ -747,23 +747,35 @@ crt_init_opt(crt_group_id_t grpid, uint32_t flags, crt_init_options_t *opt)
 			D_GOTO(unlock, rc = -DER_NONEXIST);
 		}
 
-		rc = __split_arg(interface, &iface0, &iface1);
+		/*
+		 * TODO: revisit this.
+		 * interfaces for multi-provider mode for now are delimited with ':'
+		 * to allow multi-interface selection (delimited by ',') in a single provider mode.
```

Comment:
As I was reading the code, I was wondering about this. This will probably have some ripple effects in the multiprovider branch.

Reply:
Yeah, we need to chat on what to do here. 

One option is to keep it coma separated for both cases, and determine whether it's a multi-interface or a multi-provider case based on whether its a client and server side init:

- multi-interface setup is only available on clients
- multi-provider setup is only available on servers

This will keep it contained within cart mostly, but can make it confusing to the end user.

Reply:
Yeah, I feel like that could be a bit confusing. One thing that occurred to me, maybe as we productize multiprovider, we can hide the change from the user. Right now the config file reflects Cart's format for these variables, but it doesn't have to.